src/sentry/api/endpoints/source_map_debug.py

```diff
@@ -147,4 +154,55 @@ def get(self, request: Request, project: Project, event_id: str) -> Response:
                 }
             )
 
+        release_artifacts = ReleaseFile.objects.filter(release_id=release.id)
+
+        full_matches = [
+            artifact
+            for artifact in release_artifacts
+            if urlparse(artifact.name).path == urlparts.path
+        ]
+        partial_matches = [
+            artifact
+            for artifact in release_artifacts
+            if artifact.name.endswith(urlparts.path.split("/")[-1])
+        ]
+
+        if len(full_matches) == 0:
```

Comment:
@roggenkemper This endpoint is getting rather long, can we factor out some helper methods?